### PR TITLE
Test against released versions of Cockpit

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

**Comment by mvollmer** [comment]
@petervo, the check-multi-os test adds the current dev version of cockpit to a stock centos 7 cockpit dashboard, and it succeeds.  Isn't that supposed to fail without backporting  #2425 to centos?  Could you try to extend the test to actually fail as expected?

**Comment by petervo** [comment]
centos 7 is using cockpit 0.58. The support for owned messages was implemented in 0.57 so centos is not going to exibit this bug with newer versions of cockpit. A stock fedora 22 which has version 55 should.

**Comment by mvollmer** [comment]
> centos 7 is using cockpit 0.58. The support for owned messages was implemented in 0.57 so centos is not going to exibit this bug with newer versions of cockpit.

Oh, I see.  

> A stock fedora 22 which has version 55 should.

Ok, I'll add a stock fedora-22 config.  But I think we should update it from its default repos, no?  Then it should have a new enough cockpit without the bug, I think.

In any case, testing dev against stock f22 is useful, too, of course.

**Comment by mvollmer** [comment]
The created 'stock' images are all 8 Gigs on disk.  I'll put this on top of #2430 which should reduce the size.

**Comment by mvollmer** [comment]
Depends on #2430.

File: test/check-multi-os [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            stock_b.login_and_go("dashboard", href="/dashboard/list", host=None)
+            inject_extras(stock_b)
+            wait_dashboard_addresses (stock_b, [ "localhost" ])
+
+            add_machine(stock_b, dev_m.address)
+            wait_dashboard_addresses (stock_b, [ "localhost", dev_m.address ])
+
+            add_machine(dev_b, stock_m.address)
+            dev_dashboard_addresses.append(stock_m.address)
+            wait_dashboard_addresses (dev_b, dev_dashboard_addresses)
```
Comment (stefwalter) [inline test/check-multi-os]: We need to navigate into the various machines and use a few things:

 * Use dbus ... server page?
 * Something that requires execing a process ... journal page?
 * Something that requires privilege escalation with ```{ superuser: try}``` ... docker?

**Comment by mvollmer** [comment]
> We need to navigate into the various machines and use a few things:
> * Use dbus ... server page?
>  * Something that requires execing a process ... journal page?
>  * Something that requires privilege escalation with ```{ superuser: try}``` ... docker?

That would test how the A bridge can talk to A JavaScript via the B bridge, right?

I'll also try to find things where the B JavaScript talks to the A bridge.  Getting the OS information for the dashboard?

**Comment by stefwalter** [comment]
> I'll also try to find things where the B JavaScript talks to the A bridge. Getting the OS information for the dashboard?

Yes, this uses DBus. We should make sure the host name and operating system info show up on the dashboard.

**Comment by mvollmer** [comment]
Added spawn and dbus tests.

I couldn't get actual navigation across machines working with PhantomJS.  Works ok with Firefox.  I might have been stupid, though, and will try again in a separate PR, ok?

**Comment by stefwalter** [comment]
> I couldn't get actual navigation across machines working with PhantomJS. Works ok with Firefox. I might have been stupid, though, and will try again in a separate PR, ok?

This would be very good to work out. Smells like a real bug. But another PR is ok.

**Pull request closed without merging**